libraries/rush-lib/src/api/RushCustomTipsConfiguration.ts

```diff
@@ -0,0 +1,92 @@
+import { FileSystem, ITerminal, JsonFile, JsonSchema } from '@rushstack/node-core-library';
+
+import schemaJson from '../schemas/rush-custom-tips.schema.json';
+
+/**
+ * @beta
+ */
+export interface IRushCustomTipsJson {
+  prefix?: string;
+  customTips?: IRushCustomTipItemJson[];
+}
+/**
+ * @beta
+ */
+export interface IRushCustomTipItemJson {
+  id: RushCustomTipId;
+  tip: string;
+  prefix?: string;
+  severity?: RushCustomTipSeverity;
+}
+
+/**
+ * @beta
+ */
+export enum RushCustomTipSeverity {
+  log = 0,
+  warning = 1,
+  error = 2
+}
+
+/**
+ * @beta
+ * TODO: consider making this work with the plugin (i.e., the plugins are able to define their own customizable tips)
+ */
+export type RushCustomTipId = 'PNPM_MISMATCH_DEPENDENCY' | 'ANOTHER_ID_THAT_IS_EXPOSED_TO_CUSTOM';
+
+/**
+ * @beta
+ */
+export class RushCustomTipsConfiguration {
+  private static _jsonSchema: JsonSchema = JsonSchema.fromLoadedObject(schemaJson);
+
+  private _tipMap: Map<RushCustomTipId, IRushCustomTipItemJson>;
+  private _jsonFileName: string;
+
+  public readonly configuration: Readonly<IRushCustomTipsJson>;
```

Comment:
Is this part of the public API purely for the unit test?

Reply:
Currently yes. The reason why I added this was that the `ExperimentsConfigurations.ts` also has it (in the last review, you mentioned that I could refer to that file), but I didn't put much thought into why it's public. Besides, I notice that `RushPluginsConfiguration.ts` and `ArtifactoryConfiguration.ts` all have a public API for accessing the configuration. So I thought it was a pattern, and sometime in the future, we will use it. 

Are you suggesting if the unit test is the only place using it I should change it to a private API and test it in another way?

Reply:
I think it's okay